frontend/src/components/AssignmentModal/ContactInfoCard.vue

```diff
@@ -0,0 +1,418 @@
+              .addresses[addressData.id]?.state ? ['Value is required'] : []
+            "
+          />
+        </v-col>
+        <v-col
+          v-if="addressData.id"
+          cols="12"
+          sm="1"
+        >
+          <v-btn
+            style="transform: translateY(50%);"
+            color="error"
+            density="compact"
+            icon="mdi-close"
+            @click="() => openConfirmationModal(contactData.id, 'address', addressData.id)"
+          />
+        </v-col>
+        <v-col 
+          cols="12"
+          sm="8"
+        >
+          <CustomInput
+            v-model="addressData.addressLine"
+            :rules="[rules.required]"
+            :is-required="true"
+            :label="'Address line*'"
+          />
+        </v-col>
+        <v-col 
+          cols="12"
+          sm="4"
+        >
+          <CustomInput
+            v-model="addressData.zip"
+            :rules="[rules.required, rules.zipCode]"
+            :is-required="true"
+            :label="'Zip code*'"
+            :placeholder="'XXXXX-XXXX'"
+          />
+        </v-col>
+      </v-row>
+      <v-row class="pb-7">
+        <v-col cols="12">
+          <v-btn
+            color="info"
+            variant="elevated"
+            @click="() => addInfo(contactData.id, 'address')"
+          >
+            + Add address
+          </v-btn>
+        </v-col>
+      </v-row>
+      <v-divider 
+        thickness="5"
+        color="primary"
+      />
+    </v-container>
+
+    <v-container>
+      <v-row>
+        <v-col cols="12">
+          <v-btn
+            color="primary"
+            variant="elevated"
+            @click="addContact"
+          >
+            + Add new contact
+          </v-btn>
+        </v-col>
+      </v-row>
+    </v-container>
+  </v-card-text>
+</template>
+
+<script setup lang="ts">
+  import { defineEmits, reactive, computed, onMounted } from 'vue';
+  import { storeToRefs } from 'pinia';
+
+  import { type IFormRules } from '@/utils/interfaces';
+  import { useAssignmentStore } from '@/stores/assignment';
+  import { useConfirmationStore } from '@/stores/confirmation';
+
+  import CustomInput from '@/components/UI/CustomInput.vue';
+  import CustomSelect from '@/components/UI/CustomSelect.vue';
+  import CustomAutocomplete from '@/components/UI/CustomAutocomplete.vue';
+
+  import { 
+    contactTypesLabels,
+    phoneNumberTypesLabels, 
+    addressesTypesLabels, 
+    statesCodesLabels 
+  } from '@/helpers/assignmentModal';
+
+  const { assignmentState } = storeToRefs(useAssignmentStore());
+  const { 
+    addContactData, 
+    addContactInfoData,
+    removeContactData,
+    removePhoneNumberData,
+    removeAddressData
+  } = useAssignmentStore();
+  const { setConfirmationDataAndShow } = useConfirmationStore();
+
+  const emits = defineEmits(['validate-form']);
+
+  onMounted(() => {
+    emits('validate-form');
+  })
+
+  const getContactTypesLabels = computed(

```

Comment:
Try to refactor this computed property since it is too hard to read. For cases like this, it is OK to use if...else instead of the ternary operator to make it clearer